docs/project_setup_and_build.md

```diff
@@ -0,0 +1,229 @@
+execute operations on a Docker image.
+
+To build a project you need to use the [`bigflow build`](./cli.md#building-airflow-dags) command. The documentation you are reading is also a valid BigFlow
+project. Go to the `docs` directory and run the `bigflow build` command to see how the build process works. The `bigflow build`
+command should produce three artifacts:
+
+* The `dist` directory with a Python package
+* The `image` directory with a deployment configuration and, optionally, with a Docker image as `.tar`
+* The `build` directory with JUnit test results
+
+The `bigflow build` command uses three subcommands to generate all the artifacts: [`bigflow build-package`](./cli.md#building-pip-package), [`bigflow build-image`](./cli.md#building-docker-image), [`bigflow build-dags`](./cli.md#building-dag-files).
+
+Now, let us go through each building element in detail, starting from the Python package.
+
+## Package
+
+### Project structure
+
+An example BigFlow project, with the standard structure, looks like this:
+
+```
+project_dir/
+    project_package/
+        __init__.py
+        workflow.py
+    test/
+        __init__.py
+    resources/
+        requirements.txt
+    Dockerfile
+    project_setup.py
+    deployment_config.py
+```
+
+Let us start with the `project_package`. It's the Python package which contains the processing logic of your workflows.
+It also contains `Workflow` objects, which arranges parts of your processing logic into 
+a [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) (read the [Workflow & Job chapter](./workflow-and-job.md) to learn more about workflows and jobs).
+The `project_package` is used to create a standard Python package that can be installed using `pip`.
+
+`project_setup.py` is the build script for the project. It turns the `project_package` into a `.whl` package. 
+It's based on the standard Python tool  — [setuptool](https://packaging.python.org/key_projects/#setuptools).
+
+There is also the special variable  — `PROJECT_NAME` inside `project_setup.py`. In the example project, it is 
+`PROJECT_NAME = 'project_package'`. It tells BigFlow CLI which package inside the `project_dir` is the main package with
+your processing logic and workflows.
+
+You can put your tests into the `test` package. The `bigflow build-package` command runs tests automatically, before trying to build the package.
+
+The `resources` directory contains non-Python files. That is the only directory that will be packaged
+along the `project_package` (so you can access these files after installation, from the `project_package`). Any other files
+inside the `project_directory` won't be available for the `project_package`. `resources` can't be nested. So you can't
+have a directory inside the `resources` directory.
+
+The `get_resource_absolute_path` function allows you to access files from the `resources` directory, being inside the `project_package`.
```

Comment:
nie bardzo rozumiem o co tu chodzi

Reply:
+1, może to już za mocno wchodzi w internalsy

Reply:
simplified, done